onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py

```diff
@@ -0,0 +1,474 @@
+
+    # Load model and config
+    use_auth_token = args.input == os.path.join(".")
+    l_config = LlamaConfig.from_pretrained(
+        args.model_name if use_auth_token else args.input, use_auth_token=use_auth_token
+    )
+    llama = LlamaForCausalLM.from_pretrained(
+        args.model_name if use_auth_token else args.input, use_auth_token=use_auth_token, use_cache=True
+    )
+    original_model_name = args.model_name
+    args.model_name = args.model_name.split("/")[-1]
+
+    # Export to ONNX
+    if args.use_dynamo_export:
+        logger.warning("Please ensure you have installed PyTorch, ONNX, and ONNX Script as follows.")
+        logger.warning("Step 1 - PyTorch nightly: https://pytorch.org/get-started/locally/")
+        logger.warning("Step 2 - ONNX weekly: https://pypi.org/project/onnx-weekly/")
+        logger.warning(
+            "Step 3 - ONNX Script from source: https://github.com/microsoft/onnxscript#installing-onnx-script"
+        )
+        logger.warning(
+            "Note: When you install ONNX weekly, omit `onnx` when running the first line for installing ONNX Script. \
+                        This is because you already installed `onnx-weekly` in the previous step."
+        )
+        run_dynamo_export(args, l_config, llama)
+    else:
+        run_torchscript_export(args, l_config, llama)
+
+    # Change precision of exported models if not FP32
+    decoder_model_fp32_path = os.path.join(args.output, f"{args.model_name}_decoder_model_fp32.onnx")
+    decoder_with_past_model_fp32_path = os.path.join(
+        args.output, f"{args.model_name}_decoder_with_past_model_fp32.onnx"
+    )
+
+    if args.precision == Precision.FLOAT16:
+        # Convert decoder_model.onnx to FP16
+        decoder_model_fp16_path = os.path.join(args.output, f"{args.model_name}_decoder_model_fp16.onnx")
+        model = OnnxModel(onnx.load_model(decoder_model_fp32_path, load_external_data=True))
+        model.convert_float_to_float16(keep_io_types=False, op_block_list=["If"])
+        model.save_model_to_file(decoder_model_fp16_path, use_external_data_format=True, all_tensors_to_one_file=True)
+        del model
+
+        # Convert decoder_with_past_model.onnx to FP16
+        decoder_with_past_model_fp16_path = os.path.join(
+            args.output, f"{args.model_name}_decoder_with_past_model_fp16.onnx"
+        )
+        model = OnnxModel(onnx.load_model(decoder_with_past_model_fp32_path, load_external_data=True))
+        model.convert_float_to_float16(keep_io_types=False, op_block_list=["If"])
+        model.save_model_to_file(
+            decoder_with_past_model_fp16_path, use_external_data_format=True, all_tensors_to_one_file=True
+        )
+        del model
+
+    elif args.precision == Precision.INT8:
+        # Convert decoder_model.onnx to INT8
+        decoder_model_int8_path = os.path.join(args.output, f"{args.model_name}_decoder_model_int8.onnx")
+        quantization.quantize_dynamic(
```

Comment:
We shall use SmoothQuant for int8 quantization.

Reply:
From this example, SmoothQuant requires [Intel's neural compressor package](https://github.com/intel/neural-compressor) to run. 

https://github.com/microsoft/onnxruntime-inference-examples/blob/main/quantization/language_model/llama/main.py#L260-L272

Is this a dependency that is okay to introduce?

Reply:
Yes. Without SmoothQuant, the quality will degrade so it is a must for now.
You can add a requirement.txt file in this directory.

Reply:
Added